### DISCUSSION TITLE
UserWarning: Using a target size

### DISCUSSION THREAD

**Discussion by anthonysirico** [pyg-team/pytorch_geometric#4610] [Q&A] [answered]
Should I be concerned about the following warning?
```
C:\Users\Anthony Sirico\AppData\Local\Temp\ipykernel_121128\2205290571.py:10: UserWarning: Using a target size (torch.Size([1])) that is different to the input size (torch.Size([1, 1])). This will likely lead to incorrect results due to broadcasting. Please ensure they have the same size.
  loss = criterion(out, data.y)  # Compute the loss.
```
Because my results are coming out like this:
```
Epoch: 001, Train Acc: 0.0000, Test Acc: 0.0000
Epoch: 002, Train Acc: 0.0000, Test Acc: 0.0000
Epoch: 003, Train Acc: 0.0000, Test Acc: 0.0000
```

**Comment by rusty1s** [comment, marked as answer]
I think it is good practice to ensure that shapes match and that your training does not yield any warnings. So `out = out.view(-1)` should fix that. 

Nonetheless, your low performance seems to be induced by something different.

**Reply by anthonysirico** [reply]
Do you have any examples doing any form of regression whether linear of logistic on graphs?

**Reply by rusty1s** [reply]
[Here](https://github.com/pyg-team/pytorch_geometric/blob/master/examples/qm9_nn_conv.py#L105-L116) is an example of a graph-level regression task. In general, you just need to swap out your loss function and train via, e.g., `torch.nn.functional.mse_loss`.
